src/app/account/invitation/[inviteId]/new-user-account-form.tsx

```diff
@@ -1,25 +1,34 @@
-'use client'
+/**
+ * This file defines the user interface for a new user accepting an invitation.
+ * It acts as a controller that displays either an account creation form (`SetupAccountForm`)
+ * or a success message (`Success`) upon completion.
+ *
+ * This component is rendered by `InvitationHandler` only when a user is new and not authenticated.
+ */
+'use-client'
 
 import { Flex, Button, TextInput, PasswordInput } from '@mantine/core'
 import { useForm } from '@mantine/form'
 import { FC, useState } from 'react'
 import { useMutation } from '@tanstack/react-query'
-import { onCreateAccountAction, onPendingUserLoginAction } from './create-account.action'
+import { onCreateAccountAction } from '@/server/actions/invite.actions'
 import { Title } from '@mantine/core'
 import { z } from 'zod'
 import { zodResolver } from 'mantine-form-zod-resolver'
-import { handleMutationErrorsWithForm } from '@/components/errors'
+import { handleMutationErrorsWithForm, reportError } from '@/components/errors'
 import { useAuth, useSignIn } from '@clerk/nextjs'
 import { SuccessPanel } from '@/components/panel'
 import { useRouter } from 'next/navigation'
-import { SignOutPanel } from './signout-panel'
 import { LoadingMessage } from '@/components/loading'
 
-const Success: FC = () => {
+const Success: FC<{ inviteId: string }> = ({ inviteId }) => {
     const router = useRouter()
 
     const onContinue = () => {
-        router.push('/account/mfa')
+        // Store inviteId for use after MFA setup
+        localStorage.setItem('pendingInviteId', inviteId)
+        // Redirect to generic MFA page
+        router.push(`/account/mfa/app`)
```

Comment:
instead of using localStorage, we could pass the invite id as a query parameter:
```suggestion
        router.push(`/account/mfa/app?inviteId=${inviteId}`)
```
And then in the mfa page you'd read it using `useSearchParams`

Reply:
I was thinking about that solution too. The difference with localStorage is that the email link is not needed when the mfa has not been done fully and user returns in same browser.

I can switch it to a query parameter but then the email link is mandatory or the server makes an assumption which invite is accepted (in the edge case that more than one invite is sent out to same user). Will think about that... 🤔